File: 프로그래머스/2/12900. 2 x n 타일링/2 x n 타일링.py
def solution(n):
    dp = [0 for _ in range(n)]
    dp[0], dp[1] = 1, 2
    for i in range(2,n):
        dp[i] = (dp[i-1] + dp[i-2]) % 1000000007
    
    return dp[n-1]

File: 프로그래머스/2/12900. 2 x n 타일링/2 x n 타일링.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    vector<int> dp(60001);
    int answer = 0;
    dp[0] = 1;
    dp[1] = 2;
    for (int i = 2; i < n; i++) {
        dp[i] = (dp[i-1] + dp[i-2]) % 1000000007;
    }
    return dp[n-1];
}